Trappist/src/Promact.Trappist.Web/wwwroot/app/tests/test-questions/test-questions.component.ts

```diff
@@ -137,23 +144,27 @@ export class TestQuestionsComponent implements OnInit {
                 //Every question from category are concatenated to single array which will be sent to add to test
                 this.questionsToAdd = this.questionsToAdd.concat(category.questionList);
         }
-        this.testService.addTestQuestions(this.questionsToAdd, this.testId).subscribe(response => {
-            if (response) {
-                this.openSnackBar(response.message);
-                if (this.isSaveExit) {
-                    this.loader = false;
-                    this.route.navigate(['/tests']);
+        if (this.isEditTestEnabled) {
+            this.testService.addTestQuestions(this.questionsToAdd, this.testId).subscribe(response => {
+                if (response) {
+                    this.openSnackBar(response.message);
+                    if (this.isSaveExit) {
+                        this.loader = false;
```

Comment:
if loader is false in both condition then please write outside conditions. make it condition independently.